instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/CgroupV2ContainerIdExtractor.java

```diff
@@ -41,18 +42,31 @@ Optional<String> extractContainerId() {
     if (!filesystem.isReadable(V2_CGROUP_PATH)) {
       return empty();
     }
+
+    List<String> fileAsList;
     try {
-      return filesystem
-          .lines(V2_CGROUP_PATH)
-          .filter(line -> line.contains("hostname"))
-          .flatMap(line -> Stream.of(line.split("/")))
-          .map(CONTAINER_ID_RE::matcher)
-          .filter(Matcher::matches)
-          .findFirst()
-          .map(matcher -> matcher.group(0));
-    } catch (IOException e) {
+      fileAsList = filesystem.lineList(V2_CGROUP_PATH);
+    } catch (Exception e) {
```

Comment:
I think you could catch just `IOException` as the original code did.

Reply:
ok.